### PR TITLE
[AWS] Add include_linked_accounts parameter for all metrics data streams

### PR DESCRIPTION
**Pull request by kaiyan-sheng**
## What does this PR do?

This PR is to add `include_linked_accounts` config parameter into all metrics data streams. This parameter is added into Metricbeat in https://github.com/elastic/beats/pull/35648.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Screenshots
Default looks like this:
Screenshot 2023-06-07 at 11 27 38 AM

Config _Include Linked Accounts_ is default to false for now.

Screenshot 2023-06-07 at 11 30 32 AM

With the **_Include Linked Accounts_** toggle enabled, the agent policy looks like this:
```
inputs:
  - id: aws/metrics-ec2-abc724c0-0558-11ee-bd26-592a7049951d
    name: aws-1
    revision: 2
    type: aws/metrics
    use_output: default
    meta:
      package:
        name: aws
        version: 1.41.0
    data_stream:
      namespace: default
    package_policy_id: abc724c0-0558-11ee-bd26-592a7049951d
    streams:
      - id: aws/metrics-aws.ec2_metrics-abc724c0-0558-11ee-bd26-592a7049951d
        data_stream:
          dataset: aws.ec2_metrics
          type: metrics
        access_key_id: foo
        period: 5m
        tags_filter: null
        metricsets:
          - cloudwatch
        secret_access_key: boo
        include_linked_accounts: true
        metrics:
          - name:
              - CPUUtilization
              - CPUCreditUsage
              - CPUCreditBalance
              - CPUSurplusCreditBalance
              - CPUSurplusCreditsCharged
              - StatusCheckFailed
              - StatusCheckFailed_Instance
              - StatusCheckFailed_System
            statistic:
              - Average
            namespace: AWS/EC2
            resource_type: 'ec2:instance'
          - name:
              - DiskReadBytes
              - DiskReadOps
              - DiskWriteBytes
              - DiskWriteOps
              - NetworkIn
              - NetworkPacketsIn
              - NetworkOut
              - NetworkPacketsOut
            statistic:
              - Sum
            namespace: AWS/EC2
            resource_type: 'ec2:instance'
```

I tested it with the config above and it does collect metrics from both monitoring account and linked source account:
Screenshot 2023-06-07 at 7 26 22 PM

### CODE REVIEW

File: packages/aws/_dev/build/docs/README.md [hunk 1]
```diff
@@ -55,6 +55,18 @@ The extra-charges generated by GetMetricData API calls are proportional to the f
...
+The `include_linked_accounts` parameter is used to enable the inclusion of metrics from different accounts linked to a 
+main monitoring account. By setting this parameter to true, users can gather metrics from multiple AWS accounts that are
+linked through the https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-Unified-Cross-Account.html[CloudWatch cross-account observability].
+By default, the `include_linked_accounts` parameter is set to false, meaning that only metrics from the main monitoring
+account are collected. When set to true, the parameter allows the CloudWatch ListMetrics API to include metrics from
+the monitoring account and all linked accounts in the returned data, providing a comprehensive cross-account view.
+
+*_Note_:* Enabling the `include_linked_accounts` parameter requires appropriate permissions and authentication 
+credentials for the linked accounts. Users should ensure that the necessary IAM roles and policies are properly set up 
+to establish the necessary trust and access between accounts.
```
[the remark this answers is not in the document]
Comment (kaiyan-sheng) [reply packages/aws/_dev/build/docs/README.md:68]: Good question. @girodav I think you are talking about the [permissions](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-Unified-Cross-Account-Setup.html#CloudWatch-Unified-Cross-Account-Setup-permissions) to setup cross-account right?

[the remark this answers is not in the document]
Comment (girodav) [reply packages/aws/_dev/build/docs/README.md:68]: @kaiyan-sheng yes, I meant those permissions. We can definitely rephrase it and make it clearer if it sounds confusing.

[the remark this answers is not in the document]
Comment (kaiyan-sheng) [reply packages/aws/_dev/build/docs/README.md:68]: Thanks! I can do that and update both here and in Beats :)

**Review by tommyers-elastic — APPROVED** [review]

**Review by zmoog — APPROVED** [review]
LGTM 👏

**Pull request merged**